File: summerHelpers/src/main/java/com/summer/helper/utils/CUtils.java
package com.summer.helper.utils;

import android.content.Context;

import com.tencent.stat.StatService;

import java.util.HashMap;
import java.util.Properties;

/**
 * 腾讯MTA埋点统计
 * Created by xiaqiliang on 2017/5/15.
 */

public class CUtils {

    public static HashMap<String, String> getMapWithId(String id) {
        HashMap<String, String> map = new HashMap<String, String>();
        map.put("object_id", id);
        return map;
    }

    public static HashMap<String, String> getMapWithId(String id, String userID) {
        HashMap<String, String> map = new HashMap<String, String>();
        map.put("object_id", id);
        map.put("userId", userID);
        return map;
    }

    /**
     * 数据埋点，TAG+ID
     *
     * @param tag
     */
    public static void onClick(Context context, final String tag) {
        onClick(context, tag, 0);
    }

    /**
     * 数据埋点，TAG+ID
     *
     * @param tag
     */
    public static void onClick(Context context, final String tag, String markKey, String markValue) {
        onClickFinal(context, tag, markKey, markValue);
    }

    /**
     * 数据埋点，TAG+ID
     *
     * @param tag
     * @param id
     */
    public static void onClick(Context context, final String tag, final long id) {
        onClickFinal(context, tag, "click_id", id + "");
    }

    /**
     * 数据埋点，TAG+ID
     *
     * @param tag
     * @param id
     */
    public static void onClick(Context context, final String tag, final String id) {
        onClickFinal(context, tag, "click_id", id + "");
    }

    public static void onClickFinal(final Context context, final String tag, final String mark, final String id) {

        SThread.getIntances().submit(new Runnable() {
            @Override
            public void run() {
                Properties prop = new Properties();
                if (id != null && !id.equals("0")) {
                    prop.setProperty(mark, id);
                }
                StatService.trackCustomKVEvent(context, tag, prop);
            }
        });
    }
}

File: app/src/main/java/com/summer/demo/utils/CUtils.kt
package com.summer.demo.utils

import android.content.Context
import com.summer.helper.utils.SThread
import java.util.*

/**
 * 埋点帮助类,示例用的是腾讯的
 * 在这里可以改成别的平台
 * 埋点最禁在单个页面写，因为如果有切换平台的需求就完了
 * Created by xiaqiliang on 2017/5/15.
 */

object CUtils {

    fun getMapWithId(id: String): HashMap<String, String> {
        val map = HashMap<String, String>()
        map["object_id"] = id
        return map
    }

    fun getMapWithId(id: String, userID: String): HashMap<String, String> {
        val map = HashMap<String, String>()
        map["object_id"] = id
        map["userId"] = userID
        return map
    }

    /**
     * 数据埋点，TAG+ID
     *
     * @param tag
     */
    fun onClick(context: Context, tag: String, markKey: String, markValue: String) {
        onClickFinal(context, tag, markKey, markValue)
    }

    /**
     * 数据埋点，TAG+ID
     *
     * @param tag
     * @param id
     */
    @JvmOverloads
    fun onClick(context: Context, tag: String, id: Long = 0) {
        onClickFinal(context, tag, "click_id", id.toString() + "")
    }

    /**
     * 数据埋点，TAG+ID
     *
     * @param tag
     * @param id
     */
    fun onClick(context: Context, tag: String, id: String) {
        onClickFinal(context, tag, "click_id", id + "")
    }

    fun onClickFinal(context: Context, tag: String, mark: String, id: String) {

        SThread.getIntances().submit {
            /*      Properties prop = new Properties();
                if (id != null && !id.equals("0")) {
                    prop.setProperty(mark, id);
                }
                StatService.trackCustomKVEvent(context, tag, prop);*/
        }
    }
}
/**
 * 数据埋点，TAG+ID
 *
 * @param tag
 */
